### PR TITLE
Add csp compliance through django-csp, if installed

### PR DESCRIPTION
**Pull request by jsumnerPhD**
I added a third template `django_filter_csp.html` that will be rendered if `django-csp` is installed.  It adds csp nonces to script/style tags where needed and refactors some of the js into a statically loaded iife.

Hope this is of use!

### CODE REVIEW

**Comment by silentsokolov** [comment]
Thanks for RP! I think that it useful.

File: rangefilter/static/rangefilter/iife.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+(function() {
+  'use strict';
+  django.jQuery(".admindatefilter").each(
+    function(){
+      var form_id = django.jQuery(this).find("form").attr('id').slice(0,-5);
```
Comment (silentsokolov) [inline rangefilter/static/rangefilter/iife.js]: Why `.slice(0,-5)`?

File: rangefilter/templates/rangefilter/date_filter_csp.html [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        var newScript = document.createElement("script");
+        newScript.type = "text/javascript";
+        newScript.src = url;
+        newScript.onload = resolve;
+        newScript.setAttribute("nonce", "{{ spec.request.csp_nonce }}");
+        document.head.appendChild(newScript);
+    })
+    .then(
+      function() {console.debug('Success');},
+      function() {console.debug('Failure');}
```
Comment (silentsokolov) [inline rangefilter/templates/rangefilter/date_filter_csp.html]: I think the error message should include more detail. Example: `rangefilter: failure embed script`

File: rangefilter/templates/rangefilter/date_filter_csp.html [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+django.jQuery('document').ready(function () {
+    if (!('DateTimeShortcuts' in window)) {
+        var promiseList = [];
+
+        {% for m in spec.form.js %}
+            promiseList.push(embedScript("{{ m }}"));
+        {% endfor %}
+
+        Promise.all(promiseList).then(function() {
+            console.debug('All promises resolved');
```
Comment (silentsokolov) [inline rangefilter/templates/rangefilter/date_filter_csp.html]: Debug?

File: rangefilter/static/rangefilter/iife.js (hunk 1 above)
Comment (jsumnerPhD) [reply rangefilter/static/rangefilter/iife.js]: To drop "-form" from the string and get just the variable part.

File: rangefilter/templates/rangefilter/date_filter_csp.html (hunk 2 above)
Comment (jsumnerPhD) [reply rangefilter/templates/rangefilter/date_filter_csp.html]: True.  I only used this to debug the promise chain.  We could delete the .then().

File: rangefilter/templates/rangefilter/date_filter_csp.html (hunk 3 above)
Comment (jsumnerPhD) [reply rangefilter/templates/rangefilter/date_filter_csp.html]: Again, we can delete these console logs.  I should have removed them!

File: rangefilter/templates/rangefilter/date_filter_csp.html (hunk 2 above)
Comment (jsumnerPhD) [reply rangefilter/templates/rangefilter/date_filter_csp.html]: Or I can submit a new PR with the change.

Comment (silentsokolov) [reply rangefilter/templates/rangefilter/date_filter_csp.html]: Just update the current PR. Delete .then() is better

**Pull request merged**